### ISSUE TITLE
Reorganize tests folder to match the echopype package modules structure

### ISSUE THREAD

**Issue by lsetiawan** [OSOceanAcoustics/echopype#391]
The tests folder current have test files that follows the pattern `test_$module_$in_module.py` If this gets re-organized into folders that matches the `echopype` package directory, then as the code gets larger, it's easier to follow. For example `test_convert.py` currently covers everything that is in `echopype/convert`. This makes that test file really long and can be hard to follow for newcomers. If we change this to `tests/convert/test_api.py` for example, then we know that test file if for convert api related things such as testing `open_raw` functionality.

**Comment by leewujung** [comment]
@lsetiawan : this was resolved as of v0.5.3, right?
